apps/web-portal-client/lib/components/layout/Header.jsx

```diff
@@ -1,21 +1,48 @@
-import React from 'react'
-import PropTypes from 'prop-types'
+import React, {useState} from 'react'
+
+import {useAuth} from '../../features/auth/hooks'
+
+import {Menu, MenuItem, DesktopLayout, MobileLayout} from './main-menu'
+
+import {Logo} from '../Logo'
+import {Button} from '../Button'
+import {Image} from '../Image'
+import {ImageShapes} from '../ImageShapes'
+
+export function Header() {
+  const [shouldShowMobileMenu, toggleMobileMenu] = useState(false)
+  const {logout} = useAuth()
 
-export function Header(props) {
-  const {title} = props
   return (
-    <header>
-      <div className="mx-auto py-6 px-4 sm:px-6 lg:px-8">
-        <h1 className="text-3xl leading-tight text-gray-900">{title}</h1>
-      </div>
-    </header>
-  )
-}
+    <nav className="bg-gray-800">
+      <div className="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8">
+        <div className="flex items-center justify-between h-16">
+          <div className="flex items-center">
+            <Logo />
 
-Header.propTypes = {
-  title: PropTypes.string,
-}
+            <Menu layout={DesktopLayout}>
+              <MenuItem name="Home" link="/home" />
+              <MenuItem name="Examples" link="/example" />
+              <MenuItem name="Teams" link="/team" />
+            </Menu>
+          </div>
+          <div className="flex items-center space-x-1">
+            <Button label="login" handleClick={logout} />
+            <div className="-mr-2 flex md:hidden">
+              <Button handleClick={() => toggleMobileMenu(!shouldShowMobileMenu)}>
+                <Image shape={shouldShowMobileMenu ? ImageShapes.cross : ImageShapes.sandwich} />
+              </Button>
+            </div>
+          </div>
+        </div>
+      </div>
 
-Header.defaultProps = {
-  title: '',
+      <Menu layout={MobileLayout} isShown={shouldShowMobileMenu}>
```

Comment:
It seems to generic/complext to pass `layout` as dynamic component, just having DesktopMenu and MobileMenu would be enough